posthog-core/src/index.ts

```diff
@@ -331,7 +335,12 @@ export abstract class PostHogCore {
     const fetchOptions: PostHogFetchOptions = {
       method: 'POST',
       headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify({ groups, distinct_id: distinctId, token: this.apiKey }),
+      body: JSON.stringify({
+        token: this.apiKey,
+        distinct_id: distinctId,
+        $anon_distinct_id: this.getAnonymousId(),
```

Comment:
this works, although note that we don't need to send this everytime someone reloads `/decide`. Atleast once after an `identify` event is sufficient.

I follow this in posthog-js (and set the anon ID to be sent to `undefined` afterwards).

The reason being that sending anon_id when it doesn't need to be can have a perf impact.

.. .Thinking about this a bit further, we're fine to not send these whenever `distinctId === this.getAnonymousId()`, and send them whenever they are different, since that implies an identify call happened right before.

Noice, I like this simplification.

Reply:
I think the "only need to send this once" is the tricky / confusing part. It's much harder to make it only get sent once as that requires adding extra code to clear it _after_ the decide call. If it is genuinely required I think the docs could do with saying exactly that :D 

If this is the _only_ use case for keeping the previous distinct ID around then I guess we are safe to just call `setAnonId(null)` after the decide response comes back. But then on reload the /decide call wouldn't get the anon_id which means the flag would potentially come wrong, right??

Reply:
> But then on reload the /decide call wouldn't get the anon_id which means the flag would potentially come wrong, right??

Nope, so basically whenever anon_id is sent alongside, it writes to a table to tell us to always associate this user with this given id. Any subsequent calls, with or without the anonID, give the 'right' flags.

---

Agree about it being the tricky part. Hmm.. Actually, yeah, let's keep clients simple. I can tweak the server side to remain performant by following the heuristic I mentioned above.

Reply:
Awesome then that makes total sense.